git/object_scanner.go

```diff
@@ -116,6 +125,18 @@ func (s *ObjectScanner) Type() string {
 // operation.
 func (s *ObjectScanner) Err() error { return s.err }
 
+func (s *ObjectScanner) reset() error {
+	if s.object != nil && s.object.Type == "blob" {
```

Comment:
I might advocate that we check instead whether or not this is an object that can receive `Close()`, an if so call that. Perhaps like:

```go
func (s *ObjectScanner) reset() error {
        if c, ok := s.object.(interface{
                Close() error
        }); ok && c != nil {
                if err := c.Close(); err != nil {
                        return err
                }
        }
        return nil
}
```

Reply:
Okay, that seems like a nice improvement.  I'll go ahead and make that change.